models/new_model

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+
+model_name=$1
+location_mod=$2
+
+if [[ ("$1" = "help") || ("$1" = "--help") || ($# -eq 0) ]]; then
+  echo "Usage: $0 [model_name] [location_mod]"
+  exit 0
+elif [[ -d "$model_name" ]]; then
+  echo "$model_name already exists! Please try a different model name."
+  exit 1
+elif [[ $# -eq 1 ]]; then
+  echo "Please enter a proper location! (Hint: check DART/assimilation_code/location for location modules avaliable.)"
+  exit 1
+elif [[ ! -d "../assimilation_code/location/$location_mod" ]]; then
+  echo "$location_mod is not a proper location module! Please try a different location name."
+  exit 1
+else
+  cp -R template "$model_name"
+  rm -r "$model_name"/shell_scripts "$model_name"/model_mod.nml "$model_name/readme.rst"
+  mv "$model_name/new_model.rst" "$model_name/readme.rst"
+  sed -i '' "s/template_model/$model_name/" "$model_name/readme.rst"
+  sed -i '' -e  "s/template_model/$model_name/" -e "s/template_location/$location_mod/" "$model_name/work/quickbuild.sh"
```

Comment:
I can't remember the ins and outs of sed. Does sed -i ' ' work on mac and linux?

Reply:
As a frequent sed user, I can confirm that sed -i '' does work on mac. I don't know about linux though

Reply:
Because you're working with directories, I'd replace the slash with some other character to avoid any confusion. For instance: `sed -i '' "s*template_model*$model_name*"` instead of `sed -i '' "s/template_model/$model_name/"`

Reply:
Hello! I have done some testing and I have found when I substitute the slash in the commands, they usually can't find the file and I get ``sed: test*work*quickbuild.sh: No such file or directory``.  I have tried some other characters too, but no luck so far. So, I was wondering if there is some way I am meant to allow the code to read these characters as file paths? Also, if I do get another character to work, would you like me to replace all the slashes in the code to use this characters for the rest of the script? Finally, I was able to get the ``sed -i '' ...`` to work on my computer and Cheyenne. Thank you!

Reply:
try semicolons as your separators. those often work without causing new problems.

Reply:
Ah, cool! I actually found with a bit more testing that I made an oversight by replacing the file path's slashes too. This just so happens to be the confusing I am trying to prevent, which is quite something! I guess this really shows how important this change is. I might just go with the `*`s for now too, because I feel like they look the cleanest. Thanks for your help!

Reply:
i'm glad they work, but just an fyi: i think people may avoid using asterisks as separators because they are the "regular expression" wildcard character. there are times you want to use asterisks in the patterns you are matching so then they can't also be the separators.

Reply:
Oh, ok. Then would it be best for me to change them to your suggestion of the semicolons?